internal/host/hostutil/update_host.go

```diff
@@ -84,6 +85,22 @@ func UpdateHostStageTimeout(ctx context.Context, log logrus.FieldLogger, db *gor
 	return host, nil
 }
 
+func UpdateHostAndNotify(ctx context.Context, log logrus.FieldLogger, db *gorm.DB, stream stream.Notifier, infraEnvId strfmt.UUID,
+	hostId strfmt.UUID, srcStatus string, extra ...interface{}) (*common.Host, error) {
+	host, err := UpdateHost(log, db, infraEnvId, hostId, srcStatus, extra...)
+	if err != nil {
+		return nil, err
+	}
+	if host != nil {
```

Comment:
is it possible to get `host = nil` if no error is returned from `UpdateHost`?

Reply:
What I had in mind was an empty selection (where id = 'this_id_does_not_exist'). I am not too familiar with gorm but I'd say that shouldn't be an error?

Reply:
look like there is a check for that https://github.com/rccrdpccl/assisted-service/blob/dce1b1958473c31b25aced54946a32ee2c2db16b/internal/host/hostutil/update_host.go#L90-L92 :thinking: 

Not a problem, I was wondering if `UpdateHost` guarantee if `host` is set when `err == nil`.

Reply:
Nevermind, I've just looked it up and we use `Take`, which indeed returns error for empty result.